hibernate-core/src/main/java/org/hibernate/boot/internal/InFlightMetadataCollectorImpl.java

```diff
@@ -1847,10 +1848,12 @@ public MetadataBuildingContext getBuildingContext() {
 								}
 							}
 					);
-
-					fk.setName( nameIdentifier.render( getDatabase().getJdbcEnvironment().getDialect() ) );
+				} else {
```

Comment:
To clarify... "Short of that, you could rely on the fk.getName() call already being quoted using back-ticks or JPA's double-quotes *in cases where the user wants a quoted name*"